internals/api/credential.go

```diff
@@ -146,10 +147,13 @@ func (req *CreateCredentialRequest) Validate() error {
 	}
 
 	if req.Type == CredentialTypeAWSSTS {
-		_, ok := req.Metadata[CredentialMetadataAWSRole]
+		role, ok := req.Metadata[CredentialMetadataAWSRole]
 		if !ok {
 			return ErrMissingMetadata(CredentialMetadataAWSRole, CredentialTypeAWSSTS)
 		}
+		if !bytes.Equal(req.Verifier, []byte(role)) {
+			return ErrInvalidVerifier
```

Comment:
Should this error be about the verifier or about the metadata?

Reply:
It's the combination that's invalid. I'll create a new error.